### PR TITLE
Add export for MechanismConfig

### PR DESCRIPTION
**Pull request by DeeGrant**
Work completed for issue #345
- Add export option to Mechanism
- Added tests for serializer / exporter
- Moved testing mechanism to separate file
- Bug: removed gas_phase property from AqueousEquilibrium, wasn't supported by importer

Items pushed to a later PR
- Finish Edge case testing
- Add logging
- Separate mechanism_configuration.py into smaller files

### CODE REVIEW

File: musica/mechanism_configuration.py
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/mechanism_configuration.py:1]: I like this. I've added a rough implementation of it.

File: musica/test/test_parser.py [hunk 1]
```diff
@@ -1,665 +1,26 @@
...
-    assert len(mechanism.reactions.user_defined) == 1
-    validate_user_defined(mechanism.reactions.user_defined)
-    assert mechanism.version.major == 1
-    assert mechanism.version.minor == 0
-    assert mechanism.version.patch == 0
-    assert len(mechanism.reactions) == 17
-    for reaction in mechanism.reactions:
-        assert reaction is not None
-        assert isinstance(reaction.type, _ReactionType)
+from test_util_full_mechanism import *
```
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/test/test_parser.py]: Okay, thanks for letting me know. I'll try doing so from now on.

File: musica/test/test_parser.py [hunk 2]
```diff
@@ -1,665 +1,26 @@
...
 
 
 def test_parsed_full_v1_configuration():
     parser = Parser()
-    extensions = [".yaml", ".json"]
-    for extension in extensions:
+    for extension in SUPPORTED_FILE_EXTENSIONS:
```
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/test/test_parser.py]: Makes sense. Fixed

File: musica/test/test_util_full_mechanism.py [hunk 3]
```diff
@@ -0,0 +1,642 @@
+__all__ = ['SUPPORTED_FILE_EXTENSIONS', 'MECHANISM_FULLY_DEFINED', 'validate_full_v1_mechanism']
```
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/test/test_util_full_mechanism.py]: Yeah, adding the underscore in front of the private functions fixed it.

File: musica/test/test_util_full_mechanism.py [hunk 4]
```diff
@@ -0,0 +1,642 @@
...
+                aerosol_stuff, more_aerosol_stuff],
+    phases=[gas, aqueous_aerosol, surface_reacting_phase, cloud],
+    reactions=[my_arrhenius, my_other_arrhenius, my_condensed_arrhenius,
+                my_other_condensed_arrhenius, my_troe, my_branched,
+                my_tunneling, my_surface, photo_B, condensed_photo_B,
+                my_emission, my_first_order_loss, my_aqueous_equilibrium,
+                my_wet_deposition, my_henrys_law, my_simpol_phase_transfer,
+                user_defined],
+    version=Version(1, 0, 0),
+)
```
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/test/test_util_full_mechanism.py]: get_fully_defined_mechanism() is now wrapping the object.

File: musica/test/test_util_full_mechanism.py [hunk 5]
```diff
@@ -0,0 +1,642 @@
+__all__ = ['SUPPORTED_FILE_EXTENSIONS', 'MECHANISM_FULLY_DEFINED', 'validate_full_v1_mechanism']
+
+from musica.mechanism_configuration import *
+from _musica._mechanism_configuration import _ReactionType
+
+SUPPORTED_FILE_EXTENSIONS = [".yaml", ".json"]
```
[the remark this answers is not in the document]
Comment (DeeGrant) [reply musica/test/test_util_full_mechanism.py]: Removed

File: musica/mechanism_configuration.py [hunk 6]
```diff
@@ -93,6 +96,20 @@ def __init__(
...
+            'HLC_298K_mol_m3_Pa': cls.HLC_298K_mol_m3_Pa,
+            'HLC_exponential_factor_K': cls.HLC_exponential_factor_K,
+            'diffusion_coefficient_m2_s': cls.diffusion_coefficient_m2_s,
+            'N_star': cls.N_star,
+            'molecular_weight_kg_mol': cls.molecular_weight_kg_mol,
+            'density_kg_m3': cls.density_kg_m3,
+            'tracer_type': cls.tracer_type,
+            'other_properties': cls.other_properties,
+        })
+
```
Comment (K20shores) [inline musica/mechanism_configuration.py:112]: I was expecting these keys to match what we have in our [validation](https://github.com/NCAR/MechanismConfiguration/blob/main/include/mechanism_configuration/v1/validation.hpp) file for mechanism configuration. 

Lastly everything in `other_properties` would need to be serialized out as key value pairs. The code below may not be correct python, but it's at least pseduo code showing what I'm expecting. I would also expect this on the other classes

```suggestion
    @staticmethod
    def serialize(cls):
        d =  remove_empty_keys({
            'name': cls.name,
            'HLC(298K) [mol m-3 Pa-1]': cls.HLC_298K_mol_m3_Pa,
            'HLC(298K) [mol m-3 Pa-1]': cls.HLC_exponential_factor_K,
            'diffusion coefficient [m2 s-1]': cls.diffusion_coefficient_m2_s,
            'N star': cls.N_star,
            'molecular weight [kg mol-1]': cls.molecular_weight_kg_mol,
            'density [kg m-3]': cls.density_kg_m3,
            'tracer type': cls.tracer_type,
        })
        
        for key, val in cls.other_properties:
            d[key] = val
            
        return d
```

**Comment by DeeGrant** [comment]
Resolved conflicts - reopening pull request

File: musica/test/test_serializer.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import os
+import shutil
+from musica.mechanism_configuration import Serializer, Mechanism, Parser
+from test_util_full_mechanism import get_fully_defined_mechanism, validate_full_v1_mechanism
+
+
+@pytest.fixture
+def temp_dir(tmp_path):
+    yield tmp_path
+    shutil.rmtree(tmp_path)
```
Comment (K20shores) [inline musica/test/test_serializer.py:11]:
```suggestion
```

Is this even needed? Couldn't you use `tmp_path` directly in the functions below?

**Review by K20shores — APPROVED** [review]

File: musica/test/test_serializer.py (hunk 7 above)
Comment (DeeGrant) [reply musica/test/test_serializer.py:11]: You're right. I don't remember what I was trying to accomplish with that.

**Review by boulderdaze — APPROVED** [review]
Looks good!

**Review by mattldawson — APPROVED** [review]
very nice!
